src/odemis/driver/orsay.py

```diff
@@ -2656,22 +2687,255 @@ def __init__(self, name, role, parent, **kwargs):
+        super(Dataflow, self).__init__(self)
+        self._detector = detector
+        self._fib_beam = fib_beam
+        self._datamodel = detector.parent.datamodel
+
+    def start_generate(self):
+        self._datamodel.Miss.ImageTrackingNumberScanOne.Subscribe(self._listenerImageTrackNumber)
+        self._fib_beam.imagingMode.value = True
+
+    def stop_generate(self):
+        self._fib_beam.imagingMode.value = False
+        self._datamodel.Miss.ImageTrackingNumberScanOne.Unsubscribe(self._listenerImageTrackNumber)
+
+    def get(self, asap=True):
+        if asap:
+            return self._detector.receiveLatestImage()  # Return immediately the last image available
+        else:
+            self._get_single_image_counter = 0  # Image counter to ensure the single image if fully acquired.
+            is_received = threading.Event()
+
+            def wait_single_image_acq(parameter=None, attributeName="Actual"):
+                """
+                Callback function for the Orsay server parameter "ImageTrackingNumberScanOne". Stops the acquisition
+                after acquiring a single image after the first callback. Because the ImageTrackingNumberScanOne bumps
+                at the start of a scan and because of some hardware peculiarities a full image scan can with the most
+                recent dwell time can only be guaranteed when the called is called for the second time.
+
+                :param parameter: "Required by the Orsay Server"
+                :param attributeName: "Required by the Orsay Server"
+                """
+                if parameter is None or attributeName != "Actual":
+                    return
+                # The hardware acquired only a full image if the ImageTrackingNumberScanOne has increased by at least 3
+                if self._get_single_image_counter >= 1:
+                    self._datamodel.Miss.ImageTrackingNumberScanOne.Unsubscribe(wait_single_image_acq)
+                    is_received.set()
+                    self._fib_beam.imagingMode.value = False
+                else:
+                    self._get_single_image_counter += 1
+
+            self._datamodel.Miss.ImageTrackingNumberScanOne.Subscribe(wait_single_image_acq)
+            self._fib_beam.imagingMode.value = True  # Start the acquisition
+            is_received.wait()
+            self._get_single_image_counter = None
+            return self._detector.receiveLatestImage()  # Return the acquired image
+
+    def _listenerImageTrackNumber(self, parameter=None, attributeName="Actual"):
+        """
+        Callback function for the Orsay server parameter "ImageTrackingNumberScanOne".
+        Notifies all subscribed listeners with the latest image.
+
+        :param parameter: "Required by the Orsay Server"
+        :param attributeName: "Required by the Orsay Server"
+        """
+        if parameter is None or attributeName != "Actual":
+            return
+        image = self._detector.receiveLatestImage()
```

Comment:
If this fails, the function will raise an error... and it's be lost (because it's a "listener").
=> Add a try/except around the call, to log all exceptions.